src/TLuaInterpreter.cpp

```diff
@@ -3895,6 +3895,32 @@ int TLuaInterpreter::setBackgroundColor(lua_State* L)
     return 1;
 }
 
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getBackgroundColor
+int TLuaInterpreter::getBackgroundColor(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    QColor color;
+
+    QString windowName = QStringLiteral("main");
+    int n = lua_gettop(L);
+    if (n > 0) {
+        windowName = getVerifiedString(L, __func__, 1, "console name");
+    }
+
+    if (isMain(windowName)) {
+        color = host.mpConsole->getConsoleBgColor();
+    } else {
+        color = host.getBackgroundColor(windowName);
+    }
+
+    lua_pushnumber(L, color.red());
```

Comment:
As I mentioned in the previous point - there is no handling of an invalid (non-existent) name.

Reply:
Fixed!